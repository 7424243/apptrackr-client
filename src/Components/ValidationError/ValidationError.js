import React from 'react'

function ValidationError(props) {
    
    //if props.message display it, otherwise, return nothing
    if(props.message) {
        return (
            <div className='error'>{props.message}</div>
        )
    }
    return <></>
}

export default ValidationError