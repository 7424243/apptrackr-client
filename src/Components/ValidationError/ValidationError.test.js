import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ValidationError from './ValidationError'

describe('ValidationError Component', () => {

    const props = {
        message: 'test'
    }

    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<ValidationError {...props}/>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<ValidationError {...props}/>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    
})