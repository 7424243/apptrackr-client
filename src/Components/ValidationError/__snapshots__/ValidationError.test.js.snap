// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationError Component renders the UI as expected 1`] = `
<div
  className="error"
>
  test
</div>
`;
